### DISCUSSION TITLE
Creaing mirror of network

### DISCUSSION THREAD

**Discussion by anisahb** [networkx/networkx#5098] [Q&A]
Hello, if I want to create a mirror of a network, say 5 nodes to 10 nodes, what is the best way to approach this? 

I created my mirrored network via: 

```import networkx as nx
import matplotlib.pyplot as plt

g1 = nx.read_edgelist('test2.txt', create_using=nx.Graph(), nodetype=int)

print(nx.info(g1))

nx.draw(g1, with_labels=True)
plt.show()

G = nx.disjoint_union(g1, g1)
print(len(G))

nx.draw(G, with_labels=True)
plt.show()
```

To get: 
![image](https://user-images.githubusercontent.com/82027914/134374791-ea2cf8a7-5688-4901-beb1-457a9fb0aee4.png)

Which I then converted to an edgelist by: `nx.write_edgelist(G, "graph/test2.edgelist", data=False)`

Is this a valid approach to create a mirrored copy of a network?

**Comment by rossbar** [comment]
I'm not sure what you mean by "mirrored"? If you mean "mirrored" as in "reflected" in the context of *graph drawing*, then that will come down to modifying the node positions, which can be achieved by using a layout function explicitly and storing the results.

For example, let's say you have a graph and you would like to flip **the visualization** of the graph about the line `x=0` you could do something like:

```python
>>> import networkx as nx
>>> G = nx.barbell_graph(5, 1)
>>> pos = nx.spring_layout(G, seed=0xdeadc0de)
>>> pos_refl_x = {k: v * np.array([-1, 1]) for k, v in pos.items()}
>>> fig, ax = plt.subplots(1, 2, figsize=(16, 4))
>>> nx.draw(G, pos=pos, ax=ax[0], with_labels=True)
>>> nx.draw(G, pos=pos_refl_x, ax=ax[1], with_labels=True)
```
![Figure_1](https://user-images.githubusercontent.com/1268991/134546697-55df5095-e87b-47fd-b95b-e702e3e57be5.png)

Of course, other manipulations of node positions are possible (reflections about an arbitrary line, rotations, etc.) all depending on the transformations you apply to the node positions.
